File: projects/store-or/src/main/java/com/robinfood/storeor/dtos/response/StoreResponseDTO.java
package com.robinfood.storeor.dtos.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.robinfood.storeor.dtos.CommandConfiguration.CommandConfigurationResponseDTO;
import com.robinfood.storeor.dtos.configurationposbystore.StorePosDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.List;

@AllArgsConstructor
@Builder
@NoArgsConstructor
@Getter
@Setter
public class StoreResponseDTO {

    String address;

    CityResponseDTO city;

    List<CommandConfigurationResponseDTO> commandConfiguration;

    CountryResponseDTO country;

    String currencyType;

    String currencySymbol;

    String email;

    String identification;

    String internalName;

    String location;

    StoreMultiBrandDTO multiBrand;

    String name;

    String phone;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    List<StorePosDTO> pos;

    String timezone;

    StateResponseDTO state;

    TreasureDepartmentsDTO treasuryDepartment;

    Long id;

    ZoneDTO zone;

    StoreTypeDTO storeType;

    String uuid;

    StoreIdentifierTypeDTO storeIdentifierType;

    CompanyDTO company;

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/userconfiguration/StoreResponseDTO.kt
package com.robinfood.core.dtos.userconfiguration

import javax.validation.constraints.NotNull

data class StoreResponseDTO(

    @NotNull(message = "address should not be null")
    val address: String,

    @NotNull(message = "city should not be null")
    val city: String,

    @NotNull(message = "country should not be null")
    val country: String,

    @NotNull(message = "id should not be null")
    val id: Long,

    @NotNull(message = "internalName should not be null")
    val internalName: String,

    @NotNull(message = "name should not be null")
    val name: String,

    @NotNull(message = "timeZone should not be null")
    val timeZone: String,

    @NotNull(message = "uuid should not be null")
    val uuid: String

)